pkg/service/types.go

```diff
@@ -70,3 +72,36 @@ type WeightedCluster struct {
 	ClusterName ClusterName `json:"cluster_name:omitempty"`
 	Weight      int         `json:"weight:omitempty"`
 }
+
+// Provider is an interface to be implemented by components abstracting Kubernetes, and other compute/cluster providers
+type Provider interface {
+	// GetServicesForServiceAccount returns a list of services corresponding to a service account
+	GetServicesForServiceAccount(identity.K8sServiceAccount) ([]MeshService, error)
```

Comment:
Since this interface is meant to abstract providers, we shouldn't be passing k8sServiceAccount as an argument here. I understand the existing abstraction in endpoints.Provider is broken, but we can avoid this here as a part of the newly introduced interface.

Instead, have this function accept a ServiceIdentity type, and call `ToK8sServiceAccount()` within the implementation.
```suggestion
	GetServicesForServiceAccount(identity.ServiceIdentity) ([]MeshService, error)
```

Reply:
@shashankram  agreed! See associated PR that is introducing this change #3614 #3620

Reply:
Since this is a brand new interface we can introduce the change here

Reply:
Updated